### PR TITLE
Webview definition update

### PR DESCRIPTION
**Pull request by unpluggedk**
- Use factory for creating webview controllers, 
- Create a MSIDWebviewAuthorization class to handle auth session (single auth session)
- Other minor changes.

### CODE REVIEW

File: IdentityCore/src/oauth2/MSIDOauth2Factory.h [hunk 1]
```diff
@@ -56,5 +58,8 @@
 // Those methods will come with webview work...
 // - (MSIDWebAuthHandler *)webAuthHandler;
```
Comment (oldalton) [inline IdentityCore/src/oauth2/MSIDOauth2Factory.h]: nit: can you delete those comments?

File: IdentityCore/src/oauth2/MSIDOauth2Factory.h [hunk 2]
```diff
@@ -56,5 +58,8 @@
...
 // - (MSIDAuthorizationResponse *)authorizationResponse;
+- (id<MSIDWebviewInteracting>)embeddedWebviewControllerWithRequest:(MSIDRequestParameters *)requestParams
+                                                           Webview:(WKWebView *)webview;
```
Comment (oldalton) [inline IdentityCore/src/oauth2/MSIDOauth2Factory.h]: nit: WebView lowercase?

Comment (oldalton) [inline IdentityCore/src/oauth2/MSIDOauth2Factory.h]: is this the custom passed in webview?

File: IdentityCore/src/oauth2/MSIDOauth2Factory.m [hunk 3]
```diff
@@ -266,4 +266,18 @@ - (MSIDAccount *)fillAccount:(MSIDAccount *)account
     return account;
 }
 
+#pragma mark - Webview controllers
+- (id<MSIDWebviewInteracting>)embeddedWebviewControllerWithRequest:(MSIDRequestParameters *)requestParams
+                                                           Webview:(WKWebView *)webview
+{
+    // TODO: decide returning nil or default V1
```
Comment (oldalton) [inline IdentityCore/src/oauth2/MSIDOauth2Factory.m]: should return default and not V1

File: IdentityCore/src/webview/MSIDWebviewAuthorization.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+#import "MSIDRequestParameters.h"
+
+@class WKWebView;
+@class MSIDOAuth2Factory;
+
+@protocol MSIDWebviewInteracting;
+
+@interface MSIDWebviewAuthorization : NSObject
+
++ (id<MSIDWebviewInteracting>)embeddedWebviewControllerWithRequestParameters:(MSIDRequestParameters *)parameters
```
Comment (oldalton) [inline IdentityCore/src/webview/MSIDWebviewAuthorization.h]: will this method just call factory or do something else?

File: IdentityCore/src/webview/embeddedWebview/MSIDOAuth2WebviewController.m [hunk 5]
```diff
@@ -29,9 +29,18 @@
 
 @implementation MSIDOAuth2WebviewController
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/MSIDOAuth2WebviewController.m]: nit: can we call it embedded controller to be clear?

File: IdentityCore/src/oauth2/MSIDOauth2Factory.h (hunk 2 above)
Comment (unpluggedk) [reply IdentityCore/src/oauth2/MSIDOauth2Factory.h]: yes

File: IdentityCore/src/oauth2/MSIDOauth2Factory.m (hunk 3 above)
Comment (unpluggedk) [reply IdentityCore/src/oauth2/MSIDOauth2Factory.m]: yes!

File: IdentityCore/src/webview/embeddedWebview/MSIDOAuth2WebviewController.m (hunk 5 above)
Comment (unpluggedk) [reply IdentityCore/src/webview/embeddedWebview/MSIDOAuth2WebviewController.m]: sure

**Pull request merged**